### DISCUSSION TITLE
ROADMAP

### DISCUSSION THREAD

**Discussion by francoiscrouzet** [esag-swiss/iDig-Webapp#28] [General]
# First things

- [x] Define data structure https://github.com/esag-swiss/iDig-Webapp/discussions/31
    - [x] Implement data structure
- [x] Renaming component https://github.com/esag-swiss/iDig-Webapp/issues/15
- [x] Remove commented code or work in progress
- [x] remove data/preferences.json
    - [x] remove fields and types from data/preferences. Attention ces champs sont ensuite utilisés dans d'autres endroits, surement du code mort donc pas d'erreurs. Je ne retourve plus où
    - [x] remove automatedfields, optionalautomatedfields, et les mettre dans un fichier separé
    - [x] remove Agora et Amarynthos ?

# Jean Sé

- [ ] Conflict git : https://github.com/esag-swiss/iDig-Webapp/discussions/18
- [x] Quasar setup and date picker https://github.com/esag-swiss/iDig-Webapp/issues/16
- [x] export CSV https://github.com/esag-swiss/iDig-Webapp/discussions/24

# Francois

- [ ] Have a look at input architecture
- [ ] Refactor store/api in 2 parts: preparing data, and fetching them. Solution : async in the store. Where should data be prepared / set?
- [ ] Unit test for the data structure
- [x] rethrow error dans handleError() ☐   - necessaire? ☐   - c'est pas mieux de faire un return reject(error) ☐   - .catch(() => {});

# Both 

- [x] loading spinner component (with a `isLoading` state, updated in the apiclient) https://github.com/esag-swiss/iDig-Webapp/issues/29
- [x] quasar table https://github.com/esag-swiss/iDig-Webapp/discussions/3
- [x] header disabled / changed when connected - button disconnect qui met les data à null https://github.com/esag-swiss/iDig-Webapp/discussions/19
- [x] eslint errors https://github.com/esag-swiss/iDig-Webapp/issues/22
- [ ] Reponsiveness AND/OR using a quasar drawer https://github.com/esag-swiss/iDig-Webapp/issues/17
- [x] Nice display of error message:
      ☐ nice toast instead of alert
      ☐ manage if one field is undefined 
      ☐ if all field undefined, display "erreur"
      ☐ Function to rewrite error message ("ou projet")
- [x] Modif des saisie n'est pas reactive : apres avoir sauver, ça n'affiche pas les modifs dans le tableau

# Requetes par thierry et future dev 
 
- [x] Champs: implémenter un système d'accordéon par groupe et non sur l'ensemble des champs.
- [x] Champs: préciser avec un alt qu'il s'agit des champs disponibles pour le type sélectionné.
- [x] Champs: corriger le bug des checkbox checked qui ne correspondent pas aux colonnes ( Ce serait plus simple de faire figurer à côté des en-têtes de colonne dans le tableau une croix ☒ pour supprimer un champs.)
- [x] Reset des champs à afficher dans le tableau (Trench>Identifiant>Titre) lors de changement de type. voir les possibilité de customisation utilisateur (cookies? fichier json? dans préférences?)
- [x] Colonne "menu" de gauche 'sticky' : fait avec div class="sticky-top"
- [x] Seuls les champs du fichier préférences sont affichés, quid des autres ? Prévoir celui de UUID ? Le champ Attachments (aka RelationIncludesUUID) a déjà été ajouté en prévision de l'affichage des images.
- [x] Afficher par défaut les colonnes Trench>Identifiant>Titre, le Type est superflue.
- [x] Ordre des colonnes modulable avec un drag & drop.
- [x] Largeur du tableau non contrainte. : fait avec div id="app" class="container-fluid"
- [x] Ajouter fenêtre de détails sur un clic.
- [x] Éventuellement tronquer les champs multilignes (l'intégralité peut être consultée par clic)
- [x] Aligner le texte des cellules en haut.
- [x] Tester labels seulement (et pas label) pour l'affichage des champs
- [x] Déroulement indépendant fénêtres droite et gauche
- [x] Colonnes : croix pour les virer, garder les colonnes en mémoire et les reseter, largeur reglable
- [x] Pour afficher 15 colonnes, au lieu de toutes les squeezer dans la largeur de l'écran, il faudrait donner la possibilité de scroller vers la droite.
- [x] Implémenter recherche ciblée comme iDig, ce serait utile d'implémenter des critères de recherche (*%!) et la possibilité de filtrer par champ spécifique ("Description:terre cuite"). 
- [ ] ajouter un bouton "+" pour afficher des boîtes de requête supplémentaire qui permettent de filtrer un subset des données?
- [x] Interdire l'édition sur IUUD.
- [x] Faire une "belles fiche" modulable
- [x] Petit bug, la fenêtre est figée en haut de l'écran, donc quand tu scrolles en bas de ton tableau et que tu cliques sur une entrée, tu ne vois pas le popup
- [ ] Faire un tableau de comptage par contextes
- [ ] D'ailleurs, cette configuration par chaque utilisateur des champs qu'il souhaite afficher est top, mais sera-t-il aussi possible à l'avenir de sauvegarder une "configuration ESAG" que les utilisateurs pourront charger s'ils le souhaitent?
- [x] "Trenches", trouver un moyen de regrouper les trenches par campagnes, comme dans iDig, avec le même système que implémenté pour les "Champs"
- [ ] Gérer port et http
- [x] Récupérer les pref depuis idig-server
- [x] Formulaire par défaut avec champs indiqués dans préf (comme colonne de gauche)
- [x] Formulaire ouvert dans nouvel onglet
- [x] Feature : Spinner en barre de progression : barre de progression basée sur le nombre de trenches qui reste à télécharger
- [x] Feature : sélection par années (□ AMA23, □ AMA22)
- [x] Bug ? : Le champs "relation" est toujours vide, quels que soient les types sélectionnés; est-ce parce qu'il s'agit d'un champ à valeur multiple qui n'est pas encore géré par la webapp? il faut que je creuse...

# Later

- [ ] si utilisation de quasar, supprimer bootstrap completement
- [ ] Readme https://github.com/esag-swiss/iDig-Webapp/issues/14
- [ ] find a solution for the deprecated "escape" function - semi-CRITICAL. 
    - [ ] au moins faire un check que "escape" existe, et si non, afficher une alerte

**Comment by AlZorba** [comment]
Les labels traduits compliquent le processus de recherche, mais peut-être une recherche en mode expert qui utilise les noms de champ en brut, avec un module comme [https://github.com/afineedge/vue-json-query-builder ](https://github.com/afineedge/vue-json-query-builder) pourraient faciliter la construction de requêtes complexes?
